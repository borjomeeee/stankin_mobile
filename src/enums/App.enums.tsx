export enum NetworkError {
  NONE,
  NOT_CONNECTION,
  WRONG_LOGIN,
  DOWNLOAD_ERROR,
}

export enum AppModalTypes {
  ERROR,
  WARNING,
  PUSH_UP,
}
