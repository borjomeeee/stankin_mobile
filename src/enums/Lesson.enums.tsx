export enum LessonType {
  LECTURE = 'Лекция',
  SEMINAR = 'Семинар',
  LABA = 'Лаба',
  NO_TYPE = 'без типа',
}

export enum LessonGroup {
  NONE = '',
  GROUP_A = 'А',
  GROUP_B = 'Б',
}
