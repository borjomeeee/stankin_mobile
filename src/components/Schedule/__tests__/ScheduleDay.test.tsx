import React from 'react';
import renderer from 'react-test-renderer';

import ScheduleDayComponent from '../ScheduleDay.component';

import {ILesson} from '../../../models/Lesson.model';
import {LessonType, LessonGroup} from '../../../enums/Lesson.enums';
import {shallow, mount} from 'enzyme';

describe('Schedule day tests', () => {
  const lessons: ILesson[] = [
    {
      id: '1',
      title: 'Технологии программирования',
      type: LessonType.LABA,
      groupOnLesson: LessonGroup.GROUP_A,
      groupId: '1',
      room: '0001',
      teacher: 'Иванов И.И',
      date: '01/01',
      num: 1,
    },
    {
      id: '2',
      title: 'Информационные технологии и системы',
      type: LessonType.LECTURE,
      groupOnLesson: LessonGroup.GROUP_B,
      groupId: '1',
      room: '0002',
      teacher: 'Родионов И.И',
      date: '01/01',
      num: 2,
    },
    {
      id: '3',
      title: 'Английский язык',
      type: LessonType.NO_TYPE,
      groupOnLesson: LessonGroup.NONE,
      groupId: '1',
      room: '0003',
      teacher: 'Англичанов И.И',
      date: '01/01',
      num: 3,
    },
  ];
  it('Schedule day render correctly', () => {
    const tree = renderer
      .create(<ScheduleDayComponent lessons={lessons} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Schedule day checkc correct num items', () => {
    const wrapper = mount(<ScheduleDayComponent lessons={lessons} />);

    expect(wrapper.find('ScheduleLessonComponent').length).toBe(3);
  });

  it('Schedule day empty day test', () => {
    const wrapper = shallow(<ScheduleDayComponent lessons={[]} />);

    expect(wrapper.find('ScheduleDayEmptyComponent').length).toBe(1);
  });
});
