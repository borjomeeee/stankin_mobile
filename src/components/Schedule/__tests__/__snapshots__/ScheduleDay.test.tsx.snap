// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule day tests Schedule day render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView
    ItemSeparatorComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "displayName": "Styled(View)",
        "inlineStyle": InlineStyle {
          "rules": Array [
            "
  height: 5px;
",
          ],
        },
        "render": [Function],
        "shouldForwardProp": undefined,
        "styledComponentId": "StyledNativeComponent",
        "target": [Function],
        "withComponent": [Function],
      }
    }
    data={
      Array [
        Object {
          "date": "01/01",
          "groupId": "1",
          "groupOnLesson": "А",
          "id": "1",
          "num": 1,
          "room": "0001",
          "teacher": "Иванов И.И",
          "title": "Технологии программирования",
          "type": "Лаба",
        },
        Object {
          "date": "01/01",
          "groupId": "1",
          "groupOnLesson": "Б",
          "id": "2",
          "num": 2,
          "room": "0002",
          "teacher": "Родионов И.И",
          "title": "Информационные технологии и системы",
          "type": "Лекция",
        },
        Object {
          "date": "01/01",
          "groupId": "1",
          "groupOnLesson": "",
          "id": "3",
          "num": 3,
          "room": "0003",
          "teacher": "Англичанов И.И",
          "title": "Английский язык",
          "type": "без типа",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "alignContent": "stretch",
                "borderColor": "#E4E4E4",
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingRight": 15,
                "paddingTop": 10,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "paddingRight": 15,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "Inter-Bold",
                    "fontSize": 14,
                  },
                ]
              }
            >
              8:30
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#9E9E9E",
                    "fontFamily": "Inter-Bold",
                    "fontSize": 14,
                  },
                ]
              }
            >
              10:10
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderColor": "#9E9E9E",
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "marginTop": 5,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Inter-Bold",
                      "fontSize": 12,
                    },
                  ]
                }
              >
                А
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Inter-Regular",
                    "fontSize": 16,
                  },
                ]
              }
            >
              Технологии программирования
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-end",
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "marginTop": 10,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "flexDirection": "row",
                        "marginTop": 2,
                      },
                    ]
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#444444",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "fontFamily": "Inter-Regular",
                          "marginLeft": 14,
                        },
                      ]
                    }
                  >
                    0001
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "flexDirection": "row",
                        "marginTop": 2,
                      },
                    ]
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#444444",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "fontFamily": "Inter-Regular",
                          "marginLeft": 14,
                        },
                      ]
                    }
                  >
                    Иванов И.И
                  </Text>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#444444",
                      "borderRadius": 3,
                      "paddingBottom": 4,
                      "paddingLeft": 8,
                      "paddingRight": 8,
                      "paddingTop": 4,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#ffffff",
                        "fontFamily": "Inter-Regular",
                      },
                    ]
                  }
                >
                  Лаба
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          highlighted={false}
          leadingItem={
            Object {
              "date": "01/01",
              "groupId": "1",
              "groupOnLesson": "А",
              "id": "1",
              "num": 1,
              "room": "0001",
              "teacher": "Иванов И.И",
              "title": "Технологии программирования",
              "type": "Лаба",
            }
          }
          style={
            Array [
              Object {
                "height": 5,
              },
            ]
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "alignContent": "stretch",
                "borderColor": "#E4E4E4",
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingRight": 15,
                "paddingTop": 10,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "paddingRight": 15,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "Inter-Bold",
                    "fontSize": 14,
                  },
                ]
              }
            >
              10:20
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#9E9E9E",
                    "fontFamily": "Inter-Bold",
                    "fontSize": 14,
                  },
                ]
              }
            >
              12:00
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderColor": "#9E9E9E",
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "marginTop": 5,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Inter-Bold",
                      "fontSize": 12,
                    },
                  ]
                }
              >
                Б
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Inter-Regular",
                    "fontSize": 16,
                  },
                ]
              }
            >
              Информационные технологии и системы
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-end",
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "marginTop": 10,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "flexDirection": "row",
                        "marginTop": 2,
                      },
                    ]
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#444444",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "fontFamily": "Inter-Regular",
                          "marginLeft": 14,
                        },
                      ]
                    }
                  >
                    0002
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "flexDirection": "row",
                        "marginTop": 2,
                      },
                    ]
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#444444",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "fontFamily": "Inter-Regular",
                          "marginLeft": 14,
                        },
                      ]
                    }
                  >
                    Родионов И.И
                  </Text>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#444444",
                      "borderRadius": 3,
                      "paddingBottom": 4,
                      "paddingLeft": 8,
                      "paddingRight": 8,
                      "paddingTop": 4,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#ffffff",
                        "fontFamily": "Inter-Regular",
                      },
                    ]
                  }
                >
                  Лекция
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          highlighted={false}
          leadingItem={
            Object {
              "date": "01/01",
              "groupId": "1",
              "groupOnLesson": "Б",
              "id": "2",
              "num": 2,
              "room": "0002",
              "teacher": "Родионов И.И",
              "title": "Информационные технологии и системы",
              "type": "Лекция",
            }
          }
          style={
            Array [
              Object {
                "height": 5,
              },
            ]
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "alignContent": "stretch",
                "borderColor": "#E4E4E4",
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingRight": 15,
                "paddingTop": 10,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "paddingRight": 15,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "Inter-Bold",
                    "fontSize": 14,
                  },
                ]
              }
            >
              12:20
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#9E9E9E",
                    "fontFamily": "Inter-Bold",
                    "fontSize": 14,
                  },
                ]
              }
            >
              14:00
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Inter-Regular",
                    "fontSize": 16,
                  },
                ]
              }
            >
              Английский язык
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-end",
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "marginTop": 10,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "flexDirection": "row",
                        "marginTop": 2,
                      },
                    ]
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#444444",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "fontFamily": "Inter-Regular",
                          "marginLeft": 14,
                        },
                      ]
                    }
                  >
                    0003
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "flexDirection": "row",
                        "marginTop": 2,
                      },
                    ]
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#444444",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "fontFamily": "Inter-Regular",
                          "marginLeft": 14,
                        },
                      ]
                    }
                  >
                    Англичанов И.И
                  </Text>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#444444",
                      "borderRadius": 3,
                      "paddingBottom": 4,
                      "paddingLeft": 8,
                      "paddingRight": 8,
                      "paddingTop": 4,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#ffffff",
                        "fontFamily": "Inter-Regular",
                      },
                    ]
                  }
                >
                  без типа
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
